src/main/java/org/opentripplanner/transit/model/package.md

```diff
@@ -0,0 +1,66 @@
+## Transit Model Design
+
+TODO RTM - Add framework class diagram
+
+### Services and its context
+
+The `TransitService` is the main entry point for accessing all transit model objects. It may have 
+nested services or provide read-only access to key model classes. For changing the model a 
+service context should be created(obtained from the service). The context can then be used to get a
+builder for all _aggregate roots_[1]. For simplicity, we only allow _one_ context to be active at 
+any given time. Modifying, adding and deleting entities are **not** synchronized, so if more than 
+one thread are doing updates at concurrent, then the synchronization responsibility is put on the 
+client (graph builders and RealTimeSnapshot). When all modification are complete, then changes are
+made active by calling the `commit()` on the context. Before the `commit()` the changes are not 
+visible by e.g. the routing.
+
+
+### Transit Entities
+
+All transit entities must have an ID. Transit entities ar "root" level are considered _aggregate
+roots_.
+
+#### sameAs(), equals() and hashCode()
+
+`equals()` and `hashCode` uses type and `id` only, while `sameAs` is all fields including
+`id`. The `id` is used for looking up entities in indexes and for references between entities, 
+while `sameAs()` is used for modification detection.
+
+
+#### logName() and toString()
+
+We want a sensible toString() method implementation for all classes in the transit model. The
+toString is used for testing, logging and debugging, and should not be used for any domain feature
+or exposed on the APIs.
+
+Note! The `toString()` should not include nested entities, this may lead to extensive logging and
+cyclic references.
+
+For all entities the `toString()` is implemented in `AbstractTransitEntity` in the following format:
+
+```
+<Type id logName?>
```

Comment:
Would it be possible to use another sign other than `<` and `>`?

I'm asking because in the issue report it's often interpreted as a HTML entity.

On the other hand, one could argue that the HTML report building should properly escape these values.

Reply:
Yes, good point I can change it.